src/NBitcoin/Block.cs

```diff
@@ -318,18 +303,20 @@ public Block WithOptions(NetworkOptions options)
             if ((options == NetworkOptions.None) && !this.Transactions[0].HasWitness)
                 return this;
 
-            var instance = new Block();
+            Block instance = consensusFactory.CreateBlock();
             var ms = new MemoryStream();
             var bms = new BitcoinStream(ms, true)
             {
-                TransactionOptions = options
+                TransactionOptions = options,
+                ConsensusFactory = consensusFactory
```

Comment:
Note that with Consensus having a NetworkOptions member and with both TransactionOptions and ConsensusFactory basically specifying the NetworkOptions I assume that some care must be taken as to which is getting used later on.

Reply:
yeah my plan is to remove NetworkOptions as this is now handled in consensus.
So we need to bring back TransactionOptions, this cannot go in the Network (or consensus) as its a changing parameter depending on who we are connected to.